src/vm/runners/cairo_runner.rs

```diff
@@ -824,43 +824,47 @@ impl CairoRunner {
         Ok(output)
     }
 
-    ///Writes the values hosted in the output builtin's segment
-    /// Does nothing if the output builtin is not present in the program
+    /// Writes the values hosted in the output builtin's segment.
+    /// Does nothing if the output builtin is not present in the program.
     pub fn write_output(
         &mut self,
         vm: &mut VirtualMachine,
         stdout: &mut dyn io::Write,
     ) -> Result<(), RunnerError> {
-        //If the output builtin is present it will always be the first one
-        if !vm.builtin_runners.is_empty() && vm.builtin_runners[0].0 == *"output" {
-            let builtin = &vm.builtin_runners[0].1;
-            vm.segments.compute_effective_sizes(&vm.memory);
-
-            let base = builtin.base();
-
-            // After this if block,
-            // segment_used_sizes is always Some(_)
-            if vm.segments.segment_used_sizes == None {
-                vm.segments.compute_effective_sizes(&vm.memory);
-            }
+        let builtin = vm
+            .builtin_runners
+            .iter_mut()
+            .find_map(|(k, v)| match k.as_str() {
+                "output" => Some(v),
+                _ => None,
+            });
+        let builtin = match builtin {
+            Some(x) => x,
+            _ => return Ok(()),
+        };
 
-            let segment_index: usize = base
-                .try_into()
-                .map_err(|_| RunnerError::RunnerInTemporarySegment(base))?;
-            // See previous comment, the unwrap below is safe.
-            for i in 0..vm.segments.segment_used_sizes.as_ref().unwrap()[segment_index] {
-                let value = vm
-                    .memory
-                    .get_integer(&(base, i).into())
-                    .map_err(|_| RunnerError::MemoryGet((base, i).into()))?;
-                writeln!(
-                    stdout,
-                    "{}",
-                    to_field_element(value.into_owned(), vm.prime.clone())
-                )
-                .map_err(|_| RunnerError::WriteFail)?;
-            }
+        vm.segments.compute_effective_sizes(&vm.memory);
+        let base = builtin.base();
+
+        let segment_used_sizes = vm.segments.compute_effective_sizes(&vm.memory);
```

Comment:
Why is compute_effective_sizes called twice?

Reply:
It appears I might have copy-pasted instead of cut-pasted. Fixed.